magento/magento2
# magento/magento2#7720

### Description (*)

Product repository now uses store_id if it is set.

### Fixed Issues (if relevant)

1. magento/magento2#7720: Product Repository saves attribute values for existing product always on "Default Store Level"

### Manual testing scenarios

1. Create a Product, save it via ProductRepository::save.
2. Load the Product via ProductRepository::get, pass 0 as storeId parameter.
3. Save the Product via ProductRepository::save once more.
4. Make sure catalog_product_entity_* tables contain only records with store_id=0.

### Questions or comments


### Contribution checklist (*)
